src/utils/searchViewHelper.js

```diff
@@ -32,24 +47,35 @@ const searchViewHelper = {
     // If repository is selected after search value is input, fetch search results
     if (search !== '') {
       searchViewHelper.searchByRevisionOrEmail(repository, search);
+    } else {
+      // Fetch 10 most recent revisions when repository changes
+      store.dispatch(fetchRecentRevisions(repository));
     }
-    // Fetch 10 most recent revisions when repository changes
-    store.dispatch(fetchRecentRevisions(repository));
   },
 
   handleChangeSearch(event) {
     const search = event.target.value;
     const { repository } = store.getState().search;
     store.dispatch(updateSearchValue(search));
+    store.dispatch(updateSearchResults([]));
+    store.dispatch(updateInputError(false));
+    store.dispatch(updateInputHelperText(''));
```

Comment:
The same comment can be made here. It's probably a good idea to have a dedicated action to update the search value, and the reducers for each of these state values would react to this dedicated action.

Looking at the reducer, this makes it easier to reason about which user actions will change to which values.

In general, seeing several `dispatch` commands like this is a good sign that the actions aren't logical actions.
(Of course this may still happen sometimes, rules can be broken too)

Reply:
feel free to review the updated reducers and make suggestions, I can file a bug and address them in another (smaller) patch :)